### DISCUSSION TITLE
remark-math: How could I use the $` for the inline math expression

### DISCUSSION THREAD

**Discussion by HuyAms** [remarkjs/remark#1229] [Q&A] [answered]
I use `remark-math` and `rehype-katex` in my markdown. However, the symbol `$` causes conflicts with my current markdown syntax. 

My solution is to set the `singleDollarTextMath` to false

```
const remarkMathOptions: MathOptions = {
	/**
	 * Single dollar symbol to render inline epxressions in Math often interferes with “normal” dollars in text.
	 * We turn this off so we use two-dollar symbols ($$) instead
	 */
	singleDollarTextMath: false,
};

remarkPlugins={[[remarkMath, remarkMathOptions]]}
```

This solution works. However, I have to use the double dollar symbol `$$` to start the expression while our editors would like to start the expression with $` and end it with `$ instead. How could I achieve that?

**Comment by wooorm** [comment]
Hi! I don’t understand the question.
You can use one or more dollars for math.
But one dollar often conflicts.
So you can turn it off, which you do.
But then you can’t use one dollar, which is exactly what the option does?

**Reply by HuyAms** [reply]
Hi, sorry for the wrong format

So for the inline math. I want to use this syntax: 
<img width="366" alt="image" src="https://github.com/remarkjs/remark/assets/26871154/4e439931-4917-42bc-b885-c1e70a4fdae2">

Example:
```
$`a \ne 0`$
```

Instead of a single `$`or double `$$`
```
// this causes conflict
$ a \ne 0 $

// our editors don't want $$ but they prefer $`  <math-syntax> `$
$$ a \ne 0 $$
```

**Reply by wooorm** [reply, marked as answer]
Not possible. Why do you want that? Where is this syntax from?

**Reply by HuyAms** [reply]
It's the same as the syntax from the Github markdown: https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/writing-mathematical-expressions#writing-inline-expressions

They just used this syntax and don't want to migrate to the `$$` syntax.

I can do the string replace if we don't have the option that I can change the syntax

**Reply by wooorm** [reply]
Ah, GH.
GH has a rather bad math syntax that they‘ve changed over the last, what, 2 years? a couple times... I might be interested in that if it stabilizes over the coming years. But it seems fragile to switch everything over right now.

> They just used this syntax and don't want to migrate to the $$ syntax.

Where do you get this from? `$$` works too.

**Reply by HuyAms** [reply]
> Where do you get this from? $$ works too.

Ah, editors from my org want the GH syntax and it takes some extra work for them to migrate :D

**Reply by wooorm** [reply]
I think it’s very logical that you want to follow what works on GH.
In all other cases, I’d agree: we’d do what GH does too.
But for the math... I think I’d want to wait a bit to see how it stabilizes

**Reply by HuyAms** [reply]
@wooorm Thank you so much for the answer 🙏
